Pathfinding/Main.py

```diff
@@ -0,0 +1,198 @@
+
+        self.ExampleGraphBetterName = Graph()
+        self.ExampleGraphBetterName.edges = {
+            self.A: [self.B],
+            self.B: [self.A, self.C, self.D],
+            self.C: [self.A],
+            self.D: [self.E, self.A],
+            self.E: [self.B],
+            self.F: [self.A, self.E],
+            self.G: [self.A, self.B, self.C, self.D, self.E, self.F]
+        }
+
+class Grid:
+    def __init__(self):
+        self.x = 1
+        self.y = 1
+        self.NodeList = {}
+        self.GraphGrid = Graph()
+
+    def CreateNodeGrid(self):
+        while(self.y < 101):
+            if self.x < 101:
+                self.NodeList["Node x = " + str(self.x) + " y = " + str(self.y)] = Node(str("Node x = " + str(self.x) + " y = " + str(self.y)), (self.x, self.y))
+                self.x += 1
+            else:
+                self.x = 1
+                self.y += 1
+
+class GeneralGrid:
+    def __init__(self):
+        self.x = 1
+        self.y = 1
+        self.NodeList = {}
+        self.weights = {}
+        self.GraphGrid = Graph()
+
+    def CreateNodeGrid(self, SizeX, SizeY):
+        while(self.y < SizeY):
+            if self.x < SizeX:
+                self.NodeList["Node x = " + str(self.x) + " y = " + str(self.y)] = Node(str("Node x = " + str(self.x) + " y = " + str(self.y)), (self.x, self.y))
+                self.x += 1
+            else:
+                self.x = 1
+                self.y += 1
+
+    def Neighbors(self, id):
+        Results = []
+        Pos = id.Position
+        NeighborLeft = self.NodeList.get("Node x = " + str((Pos[0] - 1)) + " y = " + str(Pos[1]))
+        NeighborRight = self.NodeList.get("Node x = " + str((Pos[0] + 1)) + " y = " + str(Pos[1]))
+        NeighborBelow = self.NodeList.get("Node x = " + str(Pos[0]) + " y = " + str(Pos[1] - 1))
+        NeighborAbove = self.NodeList.get("Node x = " + str(Pos[0]) + " y = " + str(Pos[1] + 1))
+        if isinstance(NeighborLeft, Node):
+            Results.append(NeighborLeft)
+        if isinstance(NeighborRight, Node):
+            Results.append(NeighborRight)
+        if isinstance(NeighborBelow, Node):
+            Results.append(NeighborBelow)
+        if isinstance(NeighborAbove, Node):
+            Results.append(NeighborAbove)
+        return Results
+
+    def Weight(self, Node1, Node2):
+        Cost = sqrt(((Node1.Position[0] - Node2.Position[0])**2) + ((Node1.Position[1] - Node2.Position[0])**2))
+        return self.weights.get(Node2, Cost)
+
+    def Wall(self, Node):
+        Node.wall = True
+
+
+
+#A queue class which allows us to store where we have been
+class Queue:
+        def __init__(self):
+            self.elements = collections.deque()
+
+        def empty(self):
+            return len(self.elements) == 0
+
+        def put(self, x):
+            self.elements.append(x)
+
+        def get(self):
+            return self.elements.popleft()
+
+#The a star pathfinding algorithem
```

Comment:
I think this might be breadth-first search rather than A*. I don't see a heuristic being used (for example, Euclidean distance to the ultimate goal), which is the defining feature of A*. Have you seen the computerphile videos on Dijkstra's algorithm and A*? 

https://www.youtube.com/watch?v=GazC3A4OQTE
https://www.youtube.com/watch?v=ySN5Wnu88nE